### PR TITLE
light: return if target header is invalid

### LINKED ISSUE #5120: light: if we're verifying towards invalid header, Client will exhaust all witnesses
See https://github.com/tendermint/tendermint/pull/5119#discussion_r454286648

Also, sequence function does not check if we're verifying towards the same header after replacing the primary!

### PR DESCRIPTION
**Pull request by melekes**
Closes #5120

### CODE REVIEW

File: light/client.go [hunk 1]
```diff
@@ -644,20 +644,47 @@ func (c *Client) sequence(
...
-			err := fmt.Errorf("verify adjacent from #%d to #%d failed: %w",
-				trustedHeader.Height, interimHeader.Height, err)
+			err := ErrVerificationFailed{From: trustedHeader.Height, To: interimHeader.Height, Reason: err}
 
 			switch errors.Unwrap(err).(type) {
 			case ErrInvalidHeader:
+				// If the target header is invalid, return immediately.
+				if err.To == newHeader.Height {
+					c.logger.Debug("Target header is invalid", "err", err)
+					return err
```
Comment (cmwaters) [inline light/client.go:654]: nice catch 👍

File: light/client.go [hunk 2]
```diff
@@ -755,6 +781,15 @@ func (c *Client) bisectionAgainstPrimary(
 
 	switch errors.Unwrap(err).(type) {
 	case ErrInvalidHeader:
+		// If the target header is invalid, return immediately.
+		invalidHeaderHeight := err.(ErrVerificationFailed).To
```
Comment (cmwaters) [inline light/client.go:785]: Isn't the err type `ErrInvalidHeader`? why can't you just use `err.To == newHeader.Height` like you did for sequential verification

File: light/client.go [hunk 3]
```diff
@@ -644,20 +644,47 @@ func (c *Client) sequence(
...
 					c.logger.Error("Can't replace primary", "err", replaceErr)
 					// return original error
 					return err
 				}
+
+				replacementHeader, replacementVals, fErr := c.signedHeaderAndValSetFromPrimary(newHeader.Height)
+				if fErr != nil {
+					c.logger.Error("Can't fetch header/vals from primary", "err", fErr)
+					// return original error
+					return err
```
Comment (cmwaters) [inline light/client.go:671]: Do we want to throw the same `ErrVerificationFailed` errors for these situations as well - I guess it depends on what we consider the verification process

File: light/client.go (hunk 2 above)
Comment (melekes) [reply light/client.go:785]: no, it's `error` interface

File: light/client.go (hunk 3 above)
Comment (melekes) [reply light/client.go:671]: I view replacement more as a helper process and more internal. Like the client, who's using the light client should not know we're performing replacement ideally. I.e. it's hidden from the client. The client only sees the verification errors.

Comment (cmwaters) [reply light/client.go:671]: I see that makes sense 👍

**Pull request merged**